src/collective/ifttt/actions/ifttt.py

```diff
@@ -0,0 +1,146 @@
+# -*- coding: utf-8 -*-
+from concurrent.futures import ThreadPoolExecutor
+from OFS.SimpleItem import SimpleItem
+from plone.app.contentrules import PloneMessageFactory as _
+from plone.app.contentrules.actions import ActionAddForm
+from plone.app.contentrules.actions import ActionEditForm
+from plone.app.contentrules.browser.formhelper import ContentRuleFormWrapper
+from plone.contentrules.rule.interfaces import IExecutable
+from plone.contentrules.rule.interfaces import IRuleElementData
+from zope import schema
+from zope.component import adapter
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.schema.vocabulary import SimpleTerm
+from zope.schema.vocabulary import SimpleVocabulary
+
+import logging
+
+
+logger = logging.getLogger('collective.ifttt')
+
+payload_options = SimpleVocabulary([
+    SimpleTerm(value=u'Description', title=_(u'Description of action')),
+    SimpleTerm(value=u'Username', title=_(u'Username of Editor')),
+    SimpleTerm(
+        value=u'Event start date/time', title=_(u'Event Start Date/Time')
+    ),
+])
+
+
+class IIftttTriggerAction(Interface):
+    """
+        Definition of the configuration available for a  Ifttt action
+    """
+    ifttt_name = schema.TextLine(
+        title=_(u'IFTTT event name'),
+        description=_(u'The IFTTT event name'),
+        required=False,
+    )
+
+    payload_option = schema.Choice(
+        title=_(u'Choose 2nd Payload'),
+        vocabulary=payload_options,
+        description=_(
+            u'Choose whether the 3rd payload is Description, '
+            u'Username of editor, or Event Start Date/time'
+        ),
+    )
+    payload = schema.Text(
+        title=_(u'Payload'),
+        required=False,
+        description=_(u'Provide input for chosen payload')
+    )
```

Comment:
@Shriyanshagro If iFTTT can only have 3 payload fields, why would we need the custom payload field at all?

Reply:
Basically, the 3rd payload has optional input type - {description, username of editor, event start date/time}

@skleinfeldt @tkimnguyen please comment on this.

Reply:
Sorry, I had some misunderstanding with it. It's clear now